catch by const ref instead of by value or non-const ref

NathanJPhillips opened the pull request:
This PR makes the update universally for std::string (where the contents of the whole string will be copied) and bad_alloc (where memory is already short).
These two are particularly prevalent.

NathanJPhillips committed:
Added copyright notice to fix linting error

tautschnig commented on unit/testing-utils/catch.hpp:
```diff
@@ -7326,8 +7326,9 @@ namespace Catch {
-                return msg;
+            catch(const std::string &msg)
+            {
+              return msg;
```
I'm not sure this is to be touched - AFAIK the header has been copied as-is from somewhere else?

tautschnig reviewed (commented):
Looks good except I'm not sure about catch.hpp. @thk123 Could you please clarify?

NathanJPhillips committed:
catch by const ref instead of by value or non-const ref

This PR makes the update universally for std::string (where the contents of the whole string will be copied) and bad_alloc (where memory is already short).
These two are particularly prevalent.
